src/utils/getConfigManager.ts

```diff
@@ -0,0 +1,93 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import ConfigSet from '../types/ConfigSet';
+import getOptions from './getOptions';
+import loadConfig from './loadConfig';
+import { getConfigPathCandidatesForFile } from './getConfigsForFile';
+import NormalizedPath from '../types/NormalizedPath';
+
+class ConfigManager {
+    private configSet: ConfigSet = null;
+    // The subset of configs that has been loaded
+    private partialDiscoveredConfigs: ConfigSet = {};
+    // The set of paths we have checked for configs in the filesystem
+    private discoveredPaths: Set<string> = new Set();
+
+    public get all(): ConfigSet {
+        if (this.configSet === null) {
+            this._getAllConfigs();
+        }
+        return this.configSet;
+    }
+
+    public partialConfigSetForPath(configSourcePath: NormalizedPath): ConfigSet {
+        const partialSet: ConfigSet = {};
+
+        if (this.configSet) {
+            for (let configPathCandidate of getConfigPathCandidatesForFile(configSourcePath)) {
```

Comment:
Nit: you can hoist the call to `getConfigPathCandidatesForFile`

Reply:
Also, can you add some brief comments about what you're doing in these different cases?